### ISSUE TITLE
TBG Bug report #251 (falloutboy65 on 2013-03-19): MO 12.9: BOSS result page not redirected to overwrite folder when opening in Firefox

### ISSUE THREAD

**Issue by TanninOne** [TanninOne/modorganizer#32]
Hi Tannin,

a minor, yet annoying error in 12.9:
I execute BOSS and everything seems to run fine but when it should open the result HTML page in my browser it fails due to looking for it in the original location \Skyrim\Data\Skyrim:
  "Firefox can't find the file at /D:/Games/Steam/SteamApps/common/skyrim/Data/Skyrim/BOSSlog.html."

Here is the relevant part of the log:
  [...] 
 1  DEBUG (19:32:28.0169): findfirstfileex D:\Games: Games (10) 
 2  DEBUG (19:32:28.0169): findfirstfileex D:\Games\Steam: Steam (10) 
 3  DEBUG (19:32:28.0170): findfirstfileex D:\Games\Steam\SteamApps: SteamApps (10) 
 4  DEBUG (19:32:28.0170): findfirstfileex D:\Games\Steam\SteamApps\common: common (10)
 5  DEBUG (19:32:28.0170): findfirstfileex D:\Games\Steam\SteamApps\common\skyrim: skyrim (10)
 6  DEBUG (19:32:28.0170): findfirstfileex D:\Games\Steam\SteamApps\common\skyrim\data: Data (10)
 7  DEBUG (19:32:28.0170): set current directory b: D:\Games\Steam\SteamApps\common\skyrim\Data -> D:\Games\Steam\SteamApps\common\skyrim\Data
 8  DEBUG (19:32:28.0175): findfirstfileex D:\Games\Steam\SteamApps\common\skyrim\Data\Skyrim\BOSSlog.html: BOSSlog.html (20)
 9  DEBUG (19:32:28.0175): create process (w) D:\Games\Steam\SteamApps\common\skyrim\Data\Skyrim\BOSSlog.html - .\Skyrim\BOSSlog.html  (in null)
 10 DEBUG (19:32:28.0175): first access to overwrite
 11 DEBUG (19:32:28.0176): process failed to start (193)
 12 DEBUG (19:32:28.0181): get file attributes: D:\Games\Steam\SteamApps\common\skyrim\Data\Skyrim\BOSSlog.html -> D:\Games\Steam\SteamApps\common\skyrim\ModOrganizer\overwrite\Skyrim\BOSSlog.html: 20
 13 DEBUG (19:32:28.0191): create directory C:\Users\j_dough\AppData\Local\Microsoft\Windows\Caches -> C:\Users\j_dough\AppData\Local\Microsoft\Windows\Caches
 14 DEBUG (19:32:28.0203): create process (w) D:\Internet\Mozilla Firefox\firefox.exe - "D:\Internet\Mozilla Firefox\firefox.exe" -osint -url "D:\Games\Steam\SteamApps\_
                         _common\skyrim\Data\Skyrim\BOSSlog.html" (in D:\Games\Steam\SteamApps\common\skyrim\Data)
  [...]

As per log I see that the BOSSlog.html in \overwrite\Skyrim folder is actually updated to 19:32 today.

Simply execute BOSS via MO.

configuration (since first day I used BOSS): 
Start In value: D:\Games\Steam\SteamApps\common\skyrim\ModOrganizer\overwrite

**Comment by TanninOne** [comment]
falloutboy65 on 2013-03-19: The issue was updated with the following change(s):
* This issue's description has been changed

**Comment by TanninOne** [comment]
Tannin on 2013-03-20: It seems that you're using a 64-bit firefox. Unfortunately this means that the FS virtualization can't be applied to firefox and thus the path can't be resolved.
This sucks but I'm afraid I can't fix it until I can support 64-bit applications which will probably take a while.

And no, I only received one notification. :)

**Comment by TanninOne** [comment]
falloutboy65 on 2013-03-20: Hi Tannin,
I use the same standard Firefox (always 32bit to my knowledge) on a Win7 x64. The only 64bit version of Firefox I know of is called Waterfox but I never used it on this computer.

This is the same setup as before which worked perfectly fine with the previous versions. As you can see in the log snipped my applications as well as the Steam Client (and thus Skyrim) are not located on the system drive.

The only systemdrive related line is C:\Users\j_dough\AppData\Local\Microsoft\Windows\Caches which would be the same in a 32bit OS.

I split the last line of the log as I realized it would not allow me ot scroll to the end of the code section. Maybe a FF "feature" and I added some line numbers.

I don't know the source code (obviously) and I am no programmer but this is what I read (forgive if this is complete BS :) :
 The process in started in Line 9 fails in line 11 as it does not seem to access the overwrite folder correctly (.\Skyrim\BOSSlog.html). 
 This being a relative part I would expect that it depends on the "current directory" you defined in Line 7. Well, Line 7 "redirects" onto itself, the original, real 
 folder structure (D:\Games\Steam\SteamApps\common\skyrim\Data). So take Current Directory and concatenate with the relative path and you get the path seen in Line 14 (url switch): 
   D:\Games\Steam\SteamApps\common\skyrim\Data\Skyrim\BOSSlog.html

So in summary to me it seems Line 7 is the source of the problem but I am happy to learn something new.

**Comment by TanninOne** [comment]
falloutboy65 on 2013-03-20: The issue was updated with the following change(s):
* This issue's description has been changed

**Comment by TanninOne** [comment]
Tannin on 2013-03-20: The 64bit question was a shot in the dark to explain the error message on line 11, but that isn't really relevant.

The problem here is that firefox is already open and therefore not "injected" with MOs virtual filesystem. It can't see the BOSSlog.html where it's supposed to be (at D:\Games\Steam\SteamApps\common\skyrim\Data\Skyrim\BOSSlog.html).
If you quit firefox before running boss that would probably work.
I'm afraid this is a limitation of the virtual file system we're not going to get rid of since the virtual file system is always only visible to programs started from MO (even if it's indirect).

**Comment by TanninOne** [comment]
Tannin on 2013-03-20: The issue was updated with the following change(s):
* The status has been updated, from '''New''' to '''Not a bug'''.
* This issue has been closed
